tool/tsh/common/access_request.go

```diff
@@ -406,6 +406,56 @@ func showRequestTable(cf *CLIConf, reqs []types.AccessRequest) error {
 }
 
 func onRequestSearch(cf *CLIConf) error {
+	if cf.RequestableRoles && cf.ResourceKind != "" {
+		return trace.BadParameter("only one of --kind and --roles may be specified")
+	}
+	if !cf.RequestableRoles && cf.ResourceKind == "" {
+		return trace.BadParameter("one of --kind and --roles is required")
+	}
+
+	if cf.RequestableRoles {
+		return searchRequestableRoles(cf)
+	} else {
+		return searchRequestableResources(cf)
+	}
+}
+
+func searchRequestableRoles(cf *CLIConf) error {
+	tc, err := makeClient(cf)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	var caps *types.AccessCapabilities
+	err = tc.WithRootClusterClient(cf.Context, func(clt authclient.ClientI) error {
+		caps, err = clt.GetAccessCapabilities(
```

Comment:
What's the difference between `GetAccessCapabilities` and `ListRequestableRoles`?

The latter seems more appropriate (at least in name).